src/components/ColumnHeader/ColumnHeader.tsx

```diff
@@ -70,29 +98,102 @@ export default function ColumnHeader({ columnIndex, columnName, dataReady, direc
     }
   }, [sortable, columnName, direction, orderByIndex])
 
-  return (
-    <th
-      ref={ref}
-      scope="col"
-      role="columnheader"
-      aria-sort={direction ?? (sortable ? 'none' : undefined)}
-      data-order-by-index={orderBySize !== undefined ? orderByIndex : undefined}
-      data-order-by-size={orderBySize}
-      aria-description={description}
-      // 1-based index, +1 for the row header
-      // TODO(SL): don't hardcode it, but get it from the table context
-      aria-colindex={columnIndex + 2}
-      title={description}
-      onClick={onClick}
-      style={columnStyle}
-      className={className}
-    >
-      {children}
-      <ColumnResizer
-        setWidth={setWidth}
-        onDoubleClick={autoResize}
-        width={width}
+  const handleContextMenu = useCallback((e: MouseEvent) => {
+    e.preventDefault()
+    setMenuPosition({
+      x: e.clientX,
+      y: e.clientY,
+    })
+    setShowMenu(true)
+  }, [])
+
+  const closeMenu = useCallback(() => {
+    setShowMenu(false)
+  }, [])
+
+  // Close menu when clicking outside
+  useEffect(() => {
+    if (showMenu) {
+      function handleClickOutside() {
+        setShowMenu(false)
+      }
+
+      document.addEventListener('click', handleClickOutside)
+      return () => {
+        document.removeEventListener('click', handleClickOutside)
+      }
+    }
+  }, [showMenu])
+
+  // Handle menu button click
+  const handleMenuButtonClick = useCallback((e: MouseEvent) => {
+    e.stopPropagation()
+    const rect = ref.current?.getBoundingClientRect()
+    if (rect) {
+      setMenuPosition({
+        x: e.clientX,
+        y: rect.bottom,
+      })
+      setShowMenu(true)
+    }
+  }, [])
+
+  const handleSort = useCallback((_colIndex: number, sortDirection: Direction | null) => {
+    if (onClick) {
+      // Create a synthetic event with sort direction for TableHeader
+      const event = {} as MouseEvent & { sortDirection: typeof sortDirection }
+      event.sortDirection = sortDirection
+      onClick(event)
+    }
+  }, [onClick])
+
+  function renderColumnMenu() {
+    return (
+      <ColumnMenu
+        column={description}
```

Comment:
```suggestion
        column={title}
```

I like the way the presentation element looks in the menu, but because `description` is being used by the component as just a sorting hint, I think for now we should put the title here instead.

Reply:
Hey @bleakley since `title` is optional the column header can be empty. I'll prefer title and fallback to `columnName` which is what I did here as well. https://github.com/hyparam/hightable/pull/147/files/791356181c3aef7e7a81239aa540bef29b0fb6be#diff-b3857b01d708aae34872d5099ffab482a7300250437a278bc5c6ea2bd104808eR187